plugins/restful/RestfulEntityBase.php

```diff
@@ -324,16 +324,40 @@ public function viewEntity($id) {
         }
 
         if (empty($info['formatter'])) {
-          if ($sub_wrapper instanceof EntityListWrapper) {
-            // Multiple values.
-            foreach ($sub_wrapper as $item_wrapper) {
-              $value[] = $this->getValueFromProperty($wrapper, $item_wrapper, $info, $public_field_name);
+          $translatable = field_is_translatable($this->entityType, field_info_field($info['property']));
+          if ($translatable) {
+            $optional_languages = array_keys(language_list());
+
+            foreach ($optional_languages as $lang) {
+              $cloned_wrapper = clone $wrapper;
+              $cloned_wrapper->language($lang);
+              $sub_wrapper = $info['wrapper_method_on_entity'] ? $cloned_wrapper : $cloned_wrapper->{$property};
+              $sub_wrapper = clone $sub_wrapper;
+              if ($sub_wrapper instanceof EntityListWrapper) {
+                // Multiple values.
+                foreach ($sub_wrapper as $item_wrapper) {
+                  $value[$lang][] = $this->getValueFromProperty($cloned_wrapper, $item_wrapper, $info, $public_field_name);
+                }
+              }
+              else {
+                // Single value.
+                $value[$lang] = $this->getValueFromProperty($cloned_wrapper, $sub_wrapper, $info, $public_field_name);
+              }
             }
           }
           else {
-            // Single value.
-            $value = $this->getValueFromProperty($wrapper, $sub_wrapper, $info, $public_field_name);
```

Comment:
Maybe we can always clone the wrapper - it will use a bit more memory, but code will be the same.